performant-software/neatline-spa
# add window view mode links, fix selection and geometry bugs

### What does this PR do?
- Detects presence of flags set by the SPA's containing view and shows links to move between full-window and contained views accordingly.
- Addresses UI bugs relating to record creation, selection, and editing.

### What issues does it address?
- Closes #137 
- Closes #136 
- Closes #133 

### How to test
- When using Neatline in an instance of Omeka S, the exhibits table (if viewed within the Omeka site container) should show a "Full window view" link for each exhibit. Clicking a full window view link should open the exhibit without any of the Omeka site's navigation or other containing matter.
- When viewing a Neatline exhibit within an Omeka S site, the upper-right navigation buttons should include a button to enter full window mode or, if currently viewing in full window, to return to contained view mode.
- Clicking the exhibit settings button when editing a Neatline exhibit should fully deselect the selected record.
- Editing the title or other fields for a new or existing record should not trigger any changes in the zoom/state of the map other than causing the unsaved changes notice to appear above it.
- Adding new geometry to a record should not result in errors, disappearing elements, or record deselection.
